lib/Sema/CSBindings.cpp

```diff
@@ -1631,8 +1640,22 @@ void BindingSet::dump(llvm::raw_ostream &out, unsigned indent) const {
     attributes.push_back("subtype_of_existential");
   auto literalKind = getLiteralKind();
   if (literalKind != LiteralBindingKind::None) {
-    auto literalAttrStr = ("[literal: " + getLiteralBindingKind(literalKind)
-                        + "]").str();
+    std::string literalAttrStr;
+    if (literalKind == LiteralBindingKind::Atom) {
+      if (TypeVar->getImpl().getAtomicLiteralKind().hasValue()) {
+        auto atomKind = TypeVar->getImpl().getAtomicLiteralKind().getValue();
+        auto atomLiteral = TypeVar->getImpl().getAtomLiteral(atomKind).str();
+        literalAttrStr = ("[literal: " + atomLiteral + "]");
+      }
+    } else if (literalKind == LiteralBindingKind::Collection) {
+      for (const auto &protocolKind : getProtocolKind()) {
+        auto collectionLiteral = TypeVar->getImpl().getCollectionLiteral(protocolKind);
+        literalAttrStr = ("[literal: " + collectionLiteral + " " + getLiteralBindingKind(literalKind) + "]").str();
```

Comment:
If there is more than one, looks like only the last will be as the value of `literalAttrStr` right?
It is intentional? Or do we want to append all 
```suggestion
        literalAttrStr += ("[literal: " + collectionLiteral + " " + getLiteralBindingKind(literalKind) + "]").str();
```

Reply:
I'm not sure about this one. The way `getProtocolKind()` is used here to print the collection type of a single type variable, I think it will only result in 1 kind at a time because an expression that might qualify as 2 collection kinds will be broken up further into 2 type variables, each with their own collection kind and one with a reference to the other. I don't think we want to append all.

Reply:
Ah I see, I thought that it could be more than one returned by `getProtocolKind()` so that would discard all values of `literalAttrStr` except for the last one. But if that is the case maybe we don't need the `for-loop` we could simply assert that is 1 kind only on the vector and get the front? Does this makes sense @xedin?

Reply:
It's not guaranteed to be just one and I think we need to print all of them, left a comment inline.

Reply:
It does make sense, thanks @xedin!